BedrockServer.cpp

```diff
@@ -1656,30 +1640,37 @@ void BedrockServer::_reply(unique_ptr<BedrockCommand>& command) {
     }
 }
 
-void BedrockServer::suppressCommandPort(const string& reason, bool suppress, bool manualOverride) {
-    // If we've set the manual override flag, then we'll only actually make this change if we've specified it again.
-    if (_suppressCommandPortManualOverride && !manualOverride) {
-        return;
+
+void BedrockServer::blockCommandPort(const string& reason) {
+    lock_guard<mutex> lock(_portMutex);
+    _commandPortBlockReasons.insert(reason);
+    _commandPortLikelyBlocked = true;
+    if (_commandPortBlockReasons.size() == 1) {
+        _commandPortPublic = nullptr;
+        _portPluginMap.clear();
     }
-    SINFO((suppress ? "Suppressing" : "Clearing") << " command port due to: " << reason);
+    SINFO("Blocking command port due to: " << reason << (_commandPortBlockReasons.size() > 1 ? " (already blocked)" : "") << ".");
+}
 
-    // Save the state of manual override. Note that it's set to *suppress* on purpose.
-    if (manualOverride) {
-        _suppressCommandPortManualOverride = suppress;
+void BedrockServer::unblockCommandPort(const string& reason) {
+    lock_guard<mutex> lock(_portMutex);
+    auto it = _commandPortBlockReasons.find(reason);
+    if (it == _commandPortBlockReasons.end()) {
+        SWARN("Tried to unblock command port because: " << reason << ", but it wasn't blocked for that reason!");
+    } else {
+        _commandPortBlockReasons.erase(it);
+        SINFO("Unblocking command port due to: " <<  reason << (_commandPortBlockReasons.size() > 0 ? " (blocks remaining)" : "") << ".");
```

Comment:
We aren't unblocking the command port here necessarily right? If there are more than 1 reasons and you are only removing one?

Reply:
Maybe `Removing command port block`